provisioning/standalone/vscode/README.md

```diff
@@ -0,0 +1,128 @@
+
+# Getting Started
+
+-   [Overview](#overview)
+-   [How to run](#how-to-run)
+-   [Basic Installation](#basic-version)
+-   [Custom Installation](#custom-version)
+-   [Advanced Features](#advanced-features)
+-   [Others](#how-to-make-templates-for-crownlabs)
+
+# Overview
+
+**CrownLabs** offers a set of custom **vscode** images. Every image is created for a specific use case. All images are based on the **vscode-base** image which represents the base of the project.
+To learn how to create your image, check the [**Custom Version**](#custom-version) section.
+
+# How to run
+
+Every **vscode image** can be run with **Docker**, passing the environment variables used by all standalone applications ([**check this**](../docs/gettingstarted.md#how-to-create-a-standalone-application))
+An example is: `docker run -e CROWNLABS_LISTEN_PORT=8001 -it --rm -p 8001:8001 harbor.crownlabs.polito.it/crownlabs-standalone/vscode-c-cpp`
+
+# Basic Version
+
+The **base image** contained in the `./base` folder is an image that contains  only the **basic features** of the editor.
+
+**Its usage is not recommended** for these reasons:
+
+-   It has not an **entrypoint** (but it contains the *start.sh* script, which will be the derived image's **entrypoint**, so it can be run anyway with `docker run vscode-base /start.sh`).
+-   It runs as **root** (so it cannot be launched in **CrownLabs**).
+-   It contains **sudo** and **apt**, which can be used by a user to **scale-up privileges** and to **install unwanted packages**.
+
+# Custom Version
+
+## How To
+
+-   [Set Dockerfile](#set-dockerfile)
+-   [Build Dockerfile](#build-dockerfile)
+
+### Set Dockerfile
+
+Starting from the [Base Version](#base-version) it is possible to develop a customized image.
+
+1.   **Create** a new **Dockerfile**
+2.   Inside it **includes** the base version Dockerfile using **Dockerfile+**'s `INCLUDE+` directive (see more [**here**](https://github.com/edrevo/dockerfile-plus.git))
+
+    ```Dockerfile
+    INCLUDE+ ./base/Dockerfile
+    ```
+
+3.   **Copy** the starting project files and directories to the root folder of the container
+4.   **Download**, through `apt-get`, all needed dependencies and packages and clean what is not necessary for your use case.
+5.   **Set** the **user** and **group** permissions of the previous files
+6.   *Optionally* **install code-server extensions** through the [**CLI utility**](#install-extensions-from-cli)
+7.   **Switch** to `${USER}`
+8.   **Add** `ENTRYPOINT ./start.sh ` at the end of the file.
+
+### Build Dockerfile
+
+To build an image is necessary to set the **Docker context** in `/provisioning/standalone/vscode` or another parent directory. That's because we are using `INCLUDE+` (see more [**here**](https://github.com/edrevo/dockerfile-plus.git)) and the used context must contain the files used by the **base image** Dockerfile.
```

Comment:
```suggestion
To build an image it is necessary to set the **Docker context** to `/provisioning/standalone/vscode` or another parent directory. That's because we are using `INCLUDE+` (see more [**here**](https://github.com/edrevo/dockerfile-plus.git)) and the used context must contain the files used by the **base image** Dockerfile.
```